patchwork-extensions-keybinds/src/main/java/com/patchworkmc/mixin/extensions/keybinds/MixinKeyBinding.java

```diff
@@ -0,0 +1,206 @@
+			if (keybinding != null) {
+				((KeyBindingHooks) keybinding).setPressed(pressed);
+			}
+		}
+	}
+
+	@Inject(method = "isPressed", at = @At("RETURN"), cancellable = true)
+	private void isPressed(CallbackInfoReturnable<Boolean> cir) {
+		cir.setReturnValue(cir.getReturnValueZ() && getKeyConflictContext().isActive() && getKeyModifier().isActive(getKeyConflictContext()));
+	}
+
+	@Inject(method = "method_1430", at = @At("HEAD"), cancellable = true)
+	private void compareTo(KeyBinding keyBinding, CallbackInfoReturnable<Integer> cir) {
+		if (this.getCategory().equals(keyBinding.getCategory())) {
+			cir.setReturnValue(I18n.translate(this.getId()).compareTo(I18n.translate(keyBinding.getId())));
+			return;
+		}
+
+		Integer tCat = categoryOrderMap.get(this.getCategory());
+		Integer oCat = categoryOrderMap.get(keyBinding.getCategory());
+
+		if (tCat == null && oCat != null) {
+			cir.setReturnValue(1);
+			return;
+		}
+
+		if (tCat != null && oCat == null) {
+			cir.setReturnValue(-1);
+			return;
+		}
+
+		if (tCat == null && oCat == null) {
+			cir.setReturnValue(I18n.translate(this.getCategory()).compareTo(I18n.translate(keyBinding.getCategory())));
+			return;
+		}
+
+		cir.setReturnValue(tCat.compareTo(oCat));
+	}
+
+	@Inject(method = "equals", at = @At("HEAD"), cancellable = true)
+	private void equals(KeyBinding binding, CallbackInfoReturnable<Boolean> cir) {
+		if (getKeyConflictContext().conflicts(((IForgeKeybinding) binding).getKeyConflictContext()) || ((IForgeKeybinding) binding).getKeyConflictContext().conflicts(getKeyConflictContext())) {
+			KeyModifier keyModifier = getKeyModifier();
+			KeyModifier otherKeyModifier = ((IForgeKeybinding) binding).getKeyModifier();
+
+			if (keyModifier.matches(((IForgeKeybinding) binding).getKey()) || otherKeyModifier.matches(getKey())) {
+				cir.setReturnValue(true);
+			} else if (getKey().equals(((IForgeKeybinding) binding).getKey())) {
+				// IN_GAME key contexts have a conflict when at least one modifier is NONE.
+				// For example: If you hold shift to crouch, you can still press E to open your inventory. This means that a Shift+E hotkey is in conflict with E.
+				// GUI and other key contexts do not have this limitation.
+				cir.setReturnValue(keyModifier == otherKeyModifier
+								|| (getKeyConflictContext().conflicts(KeyConflictContext.IN_GAME)
+								&& (keyModifier == KeyModifier.NONE || otherKeyModifier == KeyModifier.NONE)));
+			}
+		}
+	}
+
+	@Inject(method = "getLocalizedName()Ljava/lang/String;", at = @At("RETURN"), cancellable = true)
+	private void getLocalizedName(CallbackInfoReturnable<String> cir) {
+		cir.setReturnValue(getKeyModifier().getLocalizedComboName(keyCode, cir::getReturnValue));
+	}
+
+	@Inject(method = "isDefault", at = @At("RETURN"), cancellable = true)
+	private void isDefault(CallbackInfoReturnable<Boolean> cir) {
+		cir.setReturnValue(cir.getReturnValueZ() && getKeyModifier() == getKeyModifierDefault());
```

Comment:
In these cases it is better to encode the short-circuiting into your callback. Otherwise, you are potentially breaking every other RETURN or TAIL injector, since cancelling the function will prevent them from running after your injector.

In general, it is probably a mistake if your injector unconditionally uses a `setReturnValue`, since it's likely that you're just making an `@Overwrite` in disguise.

```suggestion
		if (getKeyModifier() != getKeyModifierDefault() {
			cir.setReturnValue(false);
		}
```